### DISCUSSION TITLE
Env targets possible variants

### DISCUSSION THREAD

**Discussion by fritzlolpro** [swc-project/swc#7909] [Q&A]
Hello!
Could you please clarify something about targets configuration in `env`
According to documentation https://swc.rs/docs/configuration/compilation#envtargets
Its possible to use `string[]` or `Map<String, Version>`

On by project I use
```      
env: {
       ...
     targets: {
        browsers: ['Chrome >= 85', 'Firefox >= 79', 'ios_saf >= 13'],
     },
 },
```

And it does work but I dont get it why.

I tried to look at swc source code and found 
```
pub enum Targets {
    Query(Query),
    EsModules(EsModules),
    Versions(Versions),
    HashMap(HashMap<String, QueryOrVersion, ahash::RandomState>),
}
```
In `swc-project/swc/crates/preset_env_base/src/query.rs`
It's probably the right place (plz tell if not) but I failled in my futher investigation.

Please help me to figure out this thing, so I will be ably to make a PR in documentation to clarify this for futher users.

**Comment by kwonoj** [comment]
https://github.com/swc-project/swc/discussions/7855
